src/utils/hooks/useVideos.js

```diff
@@ -0,0 +1,27 @@
+import { useState, useLayoutEffect } from 'react';
+
+import fetchData from '../functions/fetchData';
+
+const API_URL =
+  'https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json';
+
+function useVideos() {
+  const [videos, setVideos] = useState(null);
+
+  useLayoutEffect(() => {
+    const fetchVideos = async () => {
+      try {
+        const data = await fetchData(API_URL);
+        setVideos(data);
+      } catch (error) {
+        console.log('Error while fetching videos:', error);
+      }
+    };
+
+    fetchVideos();
+  }, []);
+
+  return videos;
+}
+
+export default useVideos;
```

Comment:
I don't believe hooks were part of this challenge, but it is good you're putting into practice what you've learned. Just keep an eye of not overdoing what is required. Sometimes you might work ahead and add code that you might not need at that time,  and later have to refactor because what you developed is not longer satisfying your current needs. Or the other way around, code that is sitting there but is not really used.